execution/src/transaction.rs

```diff
@@ -368,6 +384,41 @@ impl From<TransactionRequest> for NormalizedEthTransaction {
     }
 }
 
+// Intentionally left private to hide the serialization details
+// from users of `TransactionData`. This allows making changes to
+// `TransactionData` itself while remaining backwards compatible with
+// the serialization format.
+// The purpose of `Cow` wrapping all the data is to allow serializing
+// from a reference to `TransactionData` without cloning while also allowing
+// deserializing to `SerializableTransactionData` with owned data.
+// Data type which are `Copy` are left without `Cow` references.
+#[derive(Deserialize, Serialize)]
+enum SerializableTransactionData<'a> {
```

Comment:
For versioning, we could also keep separate private submodules in the future. And it would also allow us flexibility in ser/de implementations if we'd need them custom. I was trying to think of a leaner separation that we could have, but I guess most zero-copy approaches or even a simple `#[serde(borrow)]` would only take care of the deserialization part, without clone on write. And it's much more pleasant when the main public type doesn't have the lifetime generics.

Reply:
Yes it might make sense to factor it out into a separate module in the future. Probably next time we make a change to it we should do that.